net-libs/webkit-gtk/webkit-gtk-2.38.0-r410.ebuild

```diff
@@ -0,0 +1,249 @@
+
+pkg_pretend() {
+	if [[ ${MERGE_TYPE} != "binary" ]] ; then
+		if is-flagq "-g*" && ! is-flagq "-g*0" ; then
+			einfo "Checking for sufficient disk space to build ${PN} with debugging CFLAGS"
+			check-reqs_pkg_pretend
+		fi
+
+		if ! test-flag-CXX -std=c++17 ; then
+			die "You need at least GCC 7.3.x or Clang >= 5 for C++17-specific compiler flags"
+		fi
+	fi
+}
+
+pkg_setup() {
+	if [[ ${MERGE_TYPE} != "binary" ]] && is-flagq "-g*" && ! is-flagq "-g*0" ; then
+		check-reqs_pkg_setup
+	fi
+
+	python-any-r1_pkg_setup
+}
+
+src_prepare() {
+	cmake_src_prepare
+	gnome2_src_prepare
+}
+
+src_configure() {
+	# Respect CC, otherwise fails on prefix #395875
+	tc-export CC
+
+	# It does not compile on alpha without this in LDFLAGS
+	# https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=648761
+	use alpha && append-ldflags "-Wl,--no-relax"
+
+	# ld segfaults on ia64 with LDFLAGS --as-needed, bug #555504
+	use ia64 && append-ldflags "-Wl,--no-as-needed"
+
+	# Sigbuses on SPARC with mcpu and co., bug #???
+	use sparc && filter-flags "-mvis"
+
+	# https://bugs.webkit.org/show_bug.cgi?id=42070 , #301634
+	use ppc64 && append-flags "-mminimal-toc"
+
+	# Try to use less memory, bug #469942 (see Fedora .spec for reference)
+	# --no-keep-memory doesn't work on ia64, bug #502492
+	if ! use ia64; then
+		append-ldflags $(test-flags-CCLD "-Wl,--no-keep-memory")
+	fi
+
+	# Ruby situation is a bit complicated. See bug 513888
+	local rubyimpl
+	local ruby_interpreter=""
+	for rubyimpl in ${USE_RUBY}; do
+		if has_version -b "virtual/rubygems[ruby_targets_${rubyimpl}]"; then
+			ruby_interpreter="-DRUBY_EXECUTABLE=$(type -P ${rubyimpl})"
+		fi
+	done
+	# This will rarely occur. Only a couple of corner cases could lead us to
+	# that failure. See bug 513888
+	[[ -z $ruby_interpreter ]] && die "No suitable ruby interpreter found"
+
+	# TODO: Check Web Audio support
+	# should somehow let user select between them?
+
+	local mycmakeargs=(
+		${ruby_interpreter}
+		$(cmake_use_find_package gles2-only OpenGLES2)
+		$(cmake_use_find_package egl EGL)
+		$(cmake_use_find_package !gles2-only OpenGL)
+		-DBWRAP_EXECUTABLE:FILEPATH="${EPREFIX}"/usr/bin/bwrap # If bubblewrap[suid] then portage makes it go-r and cmake find_program fails with that
+		-DDBUS_PROXY_EXECUTABLE:FILEPATH="${EPREFIX}"/usr/bin/xdg-dbus-proxy
+		-DPORT=GTK
+		# Source/cmake/WebKitFeatures.cmake
+		-DENABLE_API_TESTS=$(usex test)
+		-DENABLE_BUBBLEWRAP_SANDBOX=$(usex seccomp)
+		-DENABLE_GAMEPAD=$(usex gamepad)
+		-DENABLE_GEOLOCATION=ON # Runtime optional (talks over dbus service)
+		-DENABLE_MINIBROWSER=$(usex examples)
+		-DENABLE_SPELLCHECK=$(usex spell)
+		-DENABLE_UNIFIED_BUILDS=$(usex jumbo-build)
+		-DENABLE_VIDEO=$(usex gstreamer)
+		-DENABLE_WEBGL=ON
+		# Supported only under ANGLE
+		-DENABLE_WEBGL2=OFF
+		-DENABLE_WEB_AUDIO=$(usex gstreamer)
+		# Source/cmake/OptionsGTK.cmake
+		-DENABLE_GLES2=$(usex gles2-only)
+		-DENABLE_DOCUMENTATION=ON
```

Comment:
And now in -r410 you are enabling documentation, but not having the necessary build dependency for it (gi-docgen).
BUT, as I've told, it shouldn't be building documentation in any version, and just install the pregenerated ones in SLOT=4.1 version (-r410) in src_install, similar to what gui-libs/gtk does.

IF you build it, then there's no reason to hide the option to build it for SLOT=4 (-r0) as well, but I strongly disagree with building anything when there's pregenerated ones available in the release tarball.